patches/net/minecraft/world/effect/MobEffectInstance.java.patch

```diff
@@ -105,15 +112,7 @@
 +            int p_323657_, int p_324205_, boolean p_324263_, boolean p_324000_, Optional<Boolean> p_323607_, Optional<MobEffectInstance.Details> p_324604_, Optional<java.util.Set<net.neoforged.neoforge.common.EffectCure>> cures
          ) {
 -            return new MobEffectInstance.Details(p_323657_, p_324205_, p_324263_, p_324000_, p_323607_.orElse(p_324000_), p_324604_);
--        }
 +            return new MobEffectInstance.Details(p_323657_, p_324205_, p_324263_, p_324000_, p_323607_.orElse(p_324000_), p_324604_, cures);
-+        }
-+
-+        // TODO: https://github.com/neoforged/NeoForge/issues/986 - Delete these once custom cure sync is fixed.
-+        @Deprecated(forRemoval = true, since = "1.20.6")
-+        private Details(int amplifier, int duration, boolean ambient, boolean showParticles, boolean showIcon, Optional<MobEffectInstance.Details> hiddenEffect) {
```

Comment:
Shouldn't this be kept and made `public` to avoid changes to the vanilla record constructor?

Reply:
`ModEffectInstance.Details` is package-private, but yes, keeping the vanilla record constructor should be better.

Reply:
Resolved in https://github.com/neoforged/NeoForge/pull/1041/commits/e71a8d4206bebeae304a4e0aa6f0168b080f3685